### PR TITLE
Feature/175 can transmission

### PR DESCRIPTION
**Pull request by NicoOhR**
### Summary
Communication from the STM32 to the Raspberry Pi (or any CAN connected device on the Communications line). Introduced a secondary queue as well as the Relay Thread, along with the necessary Mutex's. To debug the connection on the Pi there is this [repository](https://github.com/NicoOhR/GodAwfulPiCode).  It has been tested with an analog device and writes to a file accurate results.

### Remarks

Before closing the branch I'm going to make some changes to the timing, trying to get as many data points to the Pi as possible.

### Checks
- [x] Tested
- [ ] Housekeeping

### CODE REVIEW

File: Project/DAQ_System/Core/Inc/main.h [hunk 1]
```diff
@@ -84,8 +84,6 @@ void Error_Handler(void);
 #define STLK_RX_GPIO_Port GPIOD
 #define STLK_TX_Pin GPIO_PIN_9
 #define STLK_TX_GPIO_Port GPIOD
-#define USB_PowerSwitchOn_Pin GPIO_PIN_6
-#define USB_PowerSwitchOn_GPIO_Port GPIOG
```
Comment (crsz20) [inline Project/DAQ_System/Core/Inc/main.h:88]: Is the USB data logger still working properly?

It has happened to me once or twice that a pin is removed or no longer initialized after messing with CubeMX a bit. I may have accidentally toggled something that I did not mean to, but I did have to set the CubeMX settings again. In my case, the GPIO pin for the timestamp timer interrupt was not working properly

File: Project/DAQ_System/Core/Inc/stm32f4xx_it.h [hunk 2]
```diff
@@ -52,8 +52,6 @@ void MemManage_Handler(void);
 void BusFault_Handler(void);
 void UsageFault_Handler(void);
 void DebugMon_Handler(void);
-void CAN1_RX0_IRQHandler(void);
-void CAN1_RX1_IRQHandler(void);
```
Comment (crsz20) [inline Project/DAQ_System/Core/Inc/stm32f4xx_it.h]: Are interrupts for receiving CAN messages from the ECU now disabled?

File: Project/DAQ_System/Core/Src/can.c [hunk 3]
```diff
@@ -38,11 +39,11 @@ void MX_CAN1_Init(void)
...
   /* USER CODE END CAN1_Init 1 */
   hcan1.Instance = CAN1;
-  hcan1.Init.Prescaler = 24;
+  hcan1.Init.Prescaler = 12;
   hcan1.Init.Mode = CAN_MODE_NORMAL;
   hcan1.Init.SyncJumpWidth = CAN_SJW_1TQ;
-  hcan1.Init.TimeSeg1 = CAN_BS1_3TQ;
-  hcan1.Init.TimeSeg2 = CAN_BS2_3TQ;
+  hcan1.Init.TimeSeg1 = CAN_BS1_11TQ;
+  hcan1.Init.TimeSeg2 = CAN_BS2_2TQ;
```
Comment (crsz20) [inline Project/DAQ_System/Core/Src/can.c:46]: Make sure to document how you selected these parameters on the CubeMX settings

**Review by crsz20 — COMMENTED** [review]
Great stuff! I would approve it once we are sure that these changes are not affecting the DataLogger

File: Project/DAQ_System/Core/Inc/main.h (hunk 1 above)
Comment (NicoOhR) [reply Project/DAQ_System/Core/Inc/main.h:88]: I'll check on that next time im in the shop, I'm pretty sure it is. The same thing as what happened in `Project/DAQ_System/Core/Inc/stm32f4xx_it.h` might've happened here

File: Project/DAQ_System/Core/Inc/stm32f4xx_it.h (hunk 2 above)
Comment (NicoOhR) [reply Project/DAQ_System/Core/Inc/stm32f4xx_it.h]: They seem to be on this version, I think I might've misclicked or something on CubeMx, fixed

File: Project/DAQ_System/Core/Src/can.c (hunk 3 above)
Comment (NicoOhR) [reply Project/DAQ_System/Core/Src/can.c:46]: will do

**Review by crsz20 — APPROVED** [review]

**Pull request merged**